src/output.cpp

```diff
@@ -713,6 +713,16 @@ int query_int( const char *mes, ... )
     return atoi( raw_input.c_str() );
 }
 
+std::string query_string( const char *mes, ... )
```

Comment:
> bool query_int( std::string msg, bool &result ) 

A return value of `std::pair<int, bool>` looks better. It avoids the temporary bool parameter.

```C++
bool b;
const int v = query_int(..., &b);
if(v) {
    return v;
}
```
vs
```C++
const int p = query_int(...);
if(p.second) {
    return p.first;
}
```

This style is also used by the standard library, for example in `std::map<T>::insert`